app/lib/submission_builder/return_header1040.rb

```diff
@@ -16,8 +16,8 @@ def document
       tax_return = submission.tax_return
       intake = submission.intake
       client = submission.client
-      creation_security_information = client.efile_security_information
-      filing_security_information = submission.efile_security_information
+      creation_security_information = client.efile_security_informations.first
+      filing_security_information = client.efile_security_informations.last
```

Comment:
I worry about a scenario where there is just one EfileSecurityInformation for whatever reason - will that occur? If so, then this code will behave incorrectly, and our schema won't support us in identifying the incorrect behavior. I'm a person who's scared to live without strong support from my database schema!

Having said that, the scenario I'm worried about will not occur. The end-of-flow (Confirm Legal) form validates that it won't occur. 

So I can live with this.

Reply:
According to what the IRS told us, they want the information from the first time and last time the client touched the information before sending it to us. Grabbing from first and last should be fine, and if they click on confirm legal twice that's the last time they interacted with the site before they submitted, so we should, in fact, be using that object.

Reply:
Guess so!